std/testing/asserts.ts

```diff
@@ -381,13 +386,24 @@ export async function assertThrowsAsync<T = void>(
     }
     if (
       msgIncludes &&
+      typeof e.message !== "undefined" &&
       !stripColor(e.message).includes(stripColor(msgIncludes))
     ) {
       msg = `Expected error message to include "${msgIncludes}", but got "${
         e.message
       }"${msg ? `: ${msg}` : "."}`;
       throw new AssertionError(msg);
     }
+    if (
+      msgIncludes &&
+      typeof e.message === "undefined" &&
+      !stripColor(String(e)).includes(stripColor(msgIncludes))
+    ) {
+      msg = `Expected error message to include "${msgIncludes}", but got "${e}"${
+        msg ? `: ${msg}` : "."
+      }`;
+      throw new AssertionError(msg);
+    }
```

Comment:
I don't think you meant to duplicate this.

Reply:
I didn't want to abstract anything before I understood people's desired approach to solving this sort of problem.

My preference would be to abstract this message check functionality into it's own method or module.

Reply:
No abstraction necessary. I'm not saying to factor it out, but it looks like this block is an accidental repetition of the one above it?

Reply:
Oh, sorry I see. This is for handling non-`Error`s. I think if `msgIncludes` is passed and a non-`Error` is thrown, it should simply fail.

Reply:
Rejecting a Promise with a string isn't an unreasonable thing to do though.

Reply:
Yes it is. Name one case where `throw "message"` is more appropriate than `throw new Error("message")`.

Doesn't matter, the specification of the function is to check the error message. This shouldn't mean more than one thing as long as `Error::message` exists.

Reply:
Sorry I think there is some confusion here.

I was referring to `Promise.reject("some message");` Not throwing a string directly. 

I was under the impression from the documentation the method `assertThrowsAsync()` is meant to handle `Promise.reject()`. So I'd expect it to handle the promise reject string scenario in some form. We have to assume developers will do this sort of thing.

It could be as I've outlined in my code where it attempts to parse the non-Error object as a message, or it could fail gracefully with some message like "Please throw or reject a valid Error object."

Maybe the latter is a better approach, but I'd argue some sort of fix should be applied.

Reply:
> I was referring to `Promise.reject("some message");` Not throwing a string directly.

It's the same thing (sorry I should've used that example). Throwing and rejecting are similar with async and await, and the same best practice applies here.

> it could fail gracefully with some message like "Please throw or reject a valid Error object."

As is fail the assertion no matter what for thrown non-`Error`s? That's a good idea. Now that I think about it, this function returns what was thrown under type `Error` and allowing a non-`Error` to be caught and returned would violate that. Please do this.

Reply:
Will take a look into this then 👍